omc3/optics_measurements/dispersion.py

```diff
@@ -277,8 +281,12 @@ def _get_delta_columns(df, plane):
     return df
 
 
-def _is_single_dpp_bin(dpps: Sequence[float], tolerance: float = dpp.DPP_BIN_TOLERANCE) -> bool:
+def _is_single_dpp_bin(dpps: Sequence[float]) -> bool:
     """ Checks if the files would be grouped into a single dpp-bin 
     by :func:`omc3.optics_measurements.dpp._compute_ranges`. """
     # alternatively: len(omc3.optics_measurements.dpp._compute_ranges(dpps, tolerance)) == 1 
-    return np.abs(np.max(dpps) - np.min(dpps)) <= 2 * tolerance
+    if any(np.isnan(dpps)):
+        LOGGER.warning("DPPs contain NaN values. Skipping dispersion calculation.")
```

Comment:
I'm not sure I like that the helper function logs a warning about dispersion stuff, which is way out of its scope. What if we use this somewhere else?

Reply:
This warning is only issued when there are NaN dpps, which should not happen in the first place. But at least catching it here,will not try to calculate dispersion from the NaNs which will crash. This happened because of NaN in the tbt data from LEIR, which is probably anyway filtered out in the future by the changes I made for the SPS in the arc-by-arc correction branch.